packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -692,8 +692,8 @@ class ReactDOMServerRenderer {
     // find the correct previous provider based on type
     let previousProvider;
     if (this.providerIndex > -1) {
-      for (let i = 0; i <= this.providerIndex; i += 1) {
-        if (this.providerStack[i] &&
+      for (let i = this.providerIndex; i >= 0; i -= 1) {
```

Comment:
If you don't mind, `i--` would feel a bit more intuitive to me

Reply:
Done. I'm used to not allowing infix from our projects lint rules, old habits.